### ISSUE TITLE
Menubar: Menu item CSS class p-menuitem-active not being applied in Primevue v3.18.0

### ISSUE THREAD

**Issue by wisdom-17** [primefaces/primevue#3162]
### Describe the bug

Hi,

I am using the following code to apply the p-menuitem-active class to the menu item for the current page:

```
watch(
  () => route.name,
  (newRoute, previousRoute) => {
    console.log(previousRoute)
    console.log(newRoute)
    // remove active style from link to previous page
    if (items.value.find((obj) => obj.to.name === previousRoute)) {
      items.value.find((obj) => obj.to.name === previousRoute).class = ''
    }

    // add active style to link on current page
    if (items.value.find((obj) => obj.to.name === newRoute)) {
      items.value.find((obj) => obj.to.name === newRoute).class =
        'p-menuitem-active'
      console.log(items.value.find((obj) => obj.to.name === newRoute).class)
    }
  },
  // this ensures the watch is run on initialisation too
  { immediate: true }
)
```

Since updating to Primevue 3.18.0, the active style is no longer showing. I am using Vue v3.2.38

### Reproducer

https://codesandbox.io/s/dreamy-vaughan-vc964p

### PrimeVue version

3.18.0

### Vue version

3.x

### Language

ES6

### Build / Runtime

Vite

**Comment by tugcekucukoglu** [comment]
Codesandbox link does not show anything about the issue. Can not replicate the issue.

![Screen Shot 2022-10-26 at 10 27 47](https://user-images.githubusercontent.com/32294991/197964738-4efaa70d-3e54-4b34-842c-f304f9c69157.png)

**Comment by wisdom-17** [comment]
> Codesandbox link does not show anything about the issue. Can not replicate the issue.
> 
> ![Screen Shot 2022-10-26 at 10 27 47](https://user-images.githubusercontent.com/32294991/197964738-4efaa70d-3e54-4b34-842c-f304f9c69157.png)

Hi @tugcekucukoglu ,

Thanks for looking into this.

On that page (https://www.primefaces.org/primevue/menubar), click on the 'Quit' menu item. You'll see that the first item in the menu bar gets the active styling applied. I noticed this issue after upgrading to v3.18.0

**Comment by tugcekucukoglu** [comment]
Thanks for replying.

Only menuitems with submenus could have 'p-menuitem-active' class. In this demo first item has a submenu and click event opens this submenu so it gets activated. But the last item does not have a submenu so the item does not get activated.

**Comment by wisdom-17** [comment]
I appreciate your explanation but this does not explain why the first item in the menu gets activated when the last item is selected. 

In the demo (https://www.primefaces.org/primevue/menubar), clicking on 'Quit' results in the first item; 'File' being activated. Surely this is unexpected behaviour?

If you have a menubar, with items that do not have a sub menu, every time any item is clicked the first item gets the active CSS class

**Comment by cricketthomas** [comment]
I have also noticed that the first item gets highlighted when clicking link in the menu.

**Comment by mkalinski93** [comment]
I´ve compared it to the [Primereact MenuBar](https://www.primefaces.org/primereact/menubar/) after I´ve discovered the menubar in primevue. It really seems unexpected behavior to me.

**Comment by wisdom-17** [comment]
@cricketthomas , @mkalinski93 Thanks for confirming the issue. 

Anyone know how I can get this issue re-opened? Or should I create a new issue?

I'm guessing this issue won't be getting much visibility due to it being closed...

**Comment by cricketthomas** [comment]
We can create a new issue with reference to this issue, there's a button on the UI. Or more annoyingly, tag a repo owner asking them to reopen.

**Comment by pkwestm** [comment]
> I'm also just coming across this trying to implement the dropdown toggle menu
> 
> https://primefaces.org/primevue/menu
> 
> Click on the toggle button, the first item has focus by default, I'm assuming this is not expected behaviour?

Not sure if it is expected behaviour. I tried to submit a fix for the new behaviour in https://github.com/primefaces/primevue/pull/3211. There is also a workaround in attached issue to that pull.

**Comment by websanova** [comment]
Just a side note, a super quick fix to disable the focus all together (may not work for all situations).

~~~css
.p-menuitem.p-focus > .p-menuitem-content:not(:hover) {
    background: transparent !important;
}
~~~

**Comment by mertsincan** [comment]
Hi all,

@wisdom-17 
> I am using the following code to apply the p-menuitem-active class to the menu item for the current page:
- This is by design. For your case, I think you can use `p-highlight` class or define a custom class in your code.

@cricketthomas 
> I have also noticed that the first item gets highlighted when clicking link in the menu. It even happens on the doc website, I clicked "Quit" and it highlighted the first item.
- Thanks a lot for the report! Fixed in https://github.com/primefaces/primevue/issues/3352

@websanova
> Click on the toggle button, the first item has focus by default, I'm assuming this is not expected behaviour?
- This is by design. Our all menu and popup components (Exp; dropdown) always focus on the first element of panel. For your case, I think you can use `<your_menu_ref>.changeFocusedOptionIndex(<index_of_your_menuitem>)` with show callback.

**Comment by vinodkola1337** [comment]
Hi @mertsincan,

> * This is by design. Our all menu and popup components (Exp; dropdown) always focus on the first element of panel. For your case, I think you can use `<your_menu_ref>.changeFocusedOptionIndex(<index_of_your_menuitem>)` with show callback.

I tried to use changeFocusedOptionIndex as you suggested, but I am not able to set the index to '-1'. Infact, when I checked the Menu component's changeFocusedOptionIndex  code, I noticed that they are intentionally converting negative values to 0 index. 

**Reference**: https://github.com/primefaces/primevue/blob/master/components/lib/menu/Menu.vue#L226
![image](https://github.com/primefaces/primevue/assets/24287012/b8898540-4f87-47d2-98e3-549a9c930ba7)

Is there any other workaround?

**Comment by pedrohcarvalhom** [comment]
> I have also noticed that the first item gets highlighted when clicking link in the menu. It even happens on the doc website, I clicked "Quit" and it highlighted the first item.

Can you guys confirm if this issue is already fixed? I'm using primevue with nuxt (3.4.6) and I'm still facing this issue. 

I can workaround using this style code, but it doesn't feel right enough

```css
.p-menuitem.p-focus > .p-menuitem-content:not(:hover) {
    background: transparent !important;
}
```
